include/oneapi/dpl/internal/random_impl/philox_engine.h

```diff
@@ -0,0 +1,373 @@
+// -*- C++ -*-
+//===-- philox-engine.h ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Philox Engine
+
+#ifndef _ONEDPL_PHILOX_ENGINE_H
+#define _ONEDPL_PHILOX_ENGINE_H
+
+#include "random_common.h"
+#include "counter_based_engines_stuff.h"
+
+namespace oneapi
+{
+namespace dpl
+{
+
+template<typename UIntType, ::std::size_t w, ::std::size_t n, ::std::size_t r,  internal::element_type_t<UIntType> ...consts>
```

Comment:
I wanted to do that on the last stage of the review, since it makes the code less readable. I've specified that in my ["before-merge" toDo](https://github.com/oneapi-src/oneDPL/pull/1785#issue-2475206702).